ersilia/publish/inspect.py

```diff
@@ -295,6 +296,16 @@ def check_comptuational_performance(self):
             executionTime = endTime - startTime
             details += f"Execution time ({n} Prediction(s)): {executionTime} seconds. "
 
+            # Try to read and analyze tracking data
+            try:
+                tracking_file = sorted(self.root.joinpath('.ersilia/tracking').glob('*.json'))[-1]
+                with open(tracking_file) as f:
+                    tracking_data = json.load(f)
+                details += f"Memory usage ({n} predictions): {tracking_data.get('memory_percent', 'N/A')}%. "
+                details += f"CPU usage ({n} predictions): {tracking_data.get('cpu_percent', 'N/A')}%. "
+            except (IndexError, FileNotFoundError, json.JSONDecodeError) as e:
+                details += f"Warning: Could not read tracking data for {n} predictions. "
+
```

Comment:
I don't understand what's happening here. We don't really need this section at all because of the way that tracking is implemented. Once a model is served with the track flag, tracking data is collated at the end of a run and simply uploaded to an S3 bucket. It is not saved in a file like tracking.json that you are using here.